gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/transformer/RequestPreProcessor.java

```diff
@@ -109,10 +110,23 @@ public EntitiesRequest transform(
         transformFilter(attributeIdMappings, originalRequest.getFilter(), context);
     Filter filter = mergeFilters(filterToInject, transformedFilter);
 
+    // Convert the time range into a filter and set it on the request so that all downstream
+    // components needn't treat it specially.
+    String timestampAttributeId = AttributeMetadataUtil.getTimestampAttributeId(
+        attributeMetadataProvider, context, originalRequest.getEntityType());
+
+    Filter.Builder filterBuilder = Filter.newBuilder()
```

Comment:
If we are doing this here, then have you removed the code to convert the default filter in QueryServiceRequestAndResponseUtils.createQsDefaultRequestFilter()?

Reply:
That change is on the way. I wanted to raise the draft with key change to get consensus first.

Reply:
Fixed this. Please check now.